js/modules/k6/experimental/fs/registry_test.go

```diff
@@ -0,0 +1,53 @@
+package fs
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFileRegistryOpen(t *testing.T) {
+	t.Parallel()
+
+	t.Run("open succeeds", func(t *testing.T) {
+		t.Parallel()
+
+		registry := &registry{}
+		fs := newTestFs(t, func(fs afero.Fs) error {
+			return afero.WriteFile(fs, "bonjour.txt", []byte("Bonjour, le monde"), 0o644)
+		})
+
+		_, gotBeforeOk := registry.openedFiles.Load("bonjour.txt")
+		gotData, gotErr := registry.open("bonjour.txt", fs)
+		_, gotAfterOk := registry.openedFiles.Load("bonjour.txt")
+
+		assert.False(t, gotBeforeOk)
+		assert.NoError(t, gotErr)
+		assert.Equal(t, []byte("Bonjour, le monde"), gotData)
+		assert.True(t, gotAfterOk)
+	})
+
+	t.Run("double open succeeds", func(t *testing.T) {
+		t.Parallel()
+
+		registry := &registry{}
+		fs := newTestFs(t, func(fs afero.Fs) error {
+			return afero.WriteFile(fs, "bonjour.txt", []byte("Bonjour, le monde"), 0o644)
+		})
+
+		firstData, firstErr := registry.open("bonjour.txt", fs)
+		_, gotFirstOk := registry.openedFiles.Load("bonjour.txt")
+		secondData, secondErr := registry.open("bonjour.txt", fs)
+		_, gotSecondOk := registry.openedFiles.Load("bonjour.txt")
+
+		assert.True(t, gotFirstOk)
+		assert.NoError(t, firstErr)
+		assert.Equal(t, []byte("Bonjour, le monde"), firstData)
+		assert.True(t, gotSecondOk)
+		assert.NoError(t, secondErr)
+		assert.True(t, bytes.Equal(firstData, secondData))
```

Comment:
This is not enough and it isn't protecting us from doing one more copy. As you can check from: https://go.dev/play/p/NkHxud8QsNn

Reply:
```
func SamePoint(x, y []int) bool {
	return cap(x) > 0 && cap(y) > 0 && &x[0:cap(x)][cap(x)-1] == &y[0:cap(y)][cap(y)-1]
}
```

Reply:
You're right, as per [my previous comment](https://github.com/grafana/k6/pull/3165#discussion_r1283150095) the only way to really check is to verify that each []byte array's first byte points to the same address, but I found it a bit hacky and wanted to find a better alternative, but didn't get to it.

Here's an example of what I mean: https://go.dev/play/p/4V5gOGZIxek

Reply:
Interesting, I was wondering why the code you posted would work, rather than checking the first byte's pointer. Found the explanation in a Google Group:

```
The two slices overlap if they both end at the same address. Once a
slice is created, you can move its starting address forward, but never
beyond start + capacity, which is a fixed value for any slice. That
syntax reslices x and y to their full capacity and takes the address
of the last element.
```